napi/src/options.rs

```diff
@@ -0,0 +1,274 @@
+
+    /// Prefer to resolve server-relative urls as absolute paths before falling back to resolve in ResolveOptions::roots.
+    ///
+    /// Default `false`
+    pub prefer_absolute: Option<bool>,
+
+    /// A list of resolve restrictions to restrict the paths that a request can be resolved on.
+    ///
+    /// Default `[]`
+    pub restrictions: Option<Vec<Restriction>>,
+
+    /// A list of directories where requests of server-relative URLs (starting with '/') are resolved.
+    /// On non-Windows systems these requests are resolved as an absolute path first.
+    ///
+    /// Default `[]`
+    pub roots: Option<Vec<String>>,
+
+    /// Whether to resolve symlinks to their symlinked location.
+    /// When enabled, symlinked resources are resolved to their real path, not their symlinked location.
+    /// Note that this may cause module resolution to fail when using tools that symlink packages (like npm link).
+    ///
+    /// Default `true`
+    pub symlinks: Option<bool>,
+
+    /// Whether to parse [module.builtinModules](https://nodejs.org/api/module.html#modulebuiltinmodules) or not.
+    /// For example, "zlib" will throw [crate::ResolveError::Builtin] when set to true.
+    ///
+    /// Default `false`
+    pub builtin_modules: Option<bool>,
+}
+
+#[napi]
+#[derive(Debug, PartialEq, Eq)]
+pub enum EnforceExtension {
+    Auto,
+    Enabled,
+    Disabled,
+}
+
+impl EnforceExtension {
+    pub fn is_auto(&self) -> bool {
+        *self == Self::Auto
+    }
+
+    pub fn is_enabled(&self) -> bool {
+        *self == Self::Enabled
+    }
+
+    pub fn is_disabled(&self) -> bool {
+        *self == Self::Disabled
+    }
+}
+
+/// Alias Value for [ResolveOptions::alias] and [ResolveOptions::fallback].
+/// Use struct because napi don't support structured union now
+#[napi(object)]
+#[derive(Debug, Clone)]
+pub struct Restriction {
+    pub path: Option<String>,
+    pub regex: Option<String>,
+}
+
+/// Tsconfig Options
+///
+/// Derived from [tsconfig-paths-webpack-plugin](https://github.com/dividab/tsconfig-paths-webpack-plugin#options)
+
+#[napi(object)]
+#[derive(Debug, Clone)]
+pub struct TsconfigOptions {
+    /// Allows you to specify where to find the TypeScript configuration file.
+    /// You may provide
+    /// * a relative path to the configuration file. It will be resolved relative to cwd.
+    /// * an absolute path to the configuration file.
+    pub config_file: String,
+    /// Support for Typescript Project References.
+    /// For the type TsconfigReferences
+    ///
+    #[napi(ts_type = "'disabled'| 'auto' | 'Array<string>'")]
+    pub references: Either<String, Vec<String>>,
+}
+
+// #[derive(Debug, Clone)]
+// pub enum TsconfigReferences {
+//     Disabled,
+//     /// Use the `references` field from tsconfig read from `config_file`.
+//     Auto,
+//     /// Manually provided relative or absolute path.
+//     Paths(Vec<String>),
+// }
+
+impl Default for NapiResolveOptions {
+    fn default() -> Self {
```

Comment:
We need a way to sync these defaults. Presumably read it off from `ResolverOptions::default()`?